packages/scenes/src/components/layout/CSSGrid/SceneCSSGridLayout.tsx

```diff
@@ -57,126 +61,276 @@ export class SceneCSSGridLayout extends SceneObjectBase<SceneCSSGridLayoutState>
       children: state.children ?? [],
       ...state,
     });
+
+    this.onPointerDown = this.onPointerDown.bind(this);
+    this.addActivationHandler(() => {
+      const dragManager = findDragManager(this);
+      if (dragManager) {
+        this.dragManager = dragManager;
+        this.dragManager.registerLayout(this);
+      }
+    });
   }
 
   public isDraggable(): boolean {
-    return false;
+    return true;
+  }
+
+  public getDragClass() {
+    return `grid-drag-handle-${this.state.key}`;
+  }
+
+  public getDragClassCancel() {
+    return 'grid-drag-cancel';
+  }
+
+  public onPointerDown(e: React.PointerEvent, item: SceneObject) {
+    if (!this.container || this.cannotDrag(e.target)) {
+      return;
+    }
+
+    e.preventDefault();
+    e.stopPropagation();
+
+    this.dragManager.onDragStart(e.nativeEvent, this, item);
+  }
+
+  private cannotDrag(el: EventTarget) {
+    const dragCancelClass = this.getDragClassCancel();
+    const dragClass = this.getDragClass();
+
+    // cancel dragging if the element being interacted with has an ancestor with the drag cancel class set
+    // or if the drag class isn't set on an ancestor
+    const cannotDrag =
+      el instanceof Element &&
+      (el.classList.contains(dragCancelClass) ||
+        el.matches(`.${dragCancelClass} *`) ||
+        (!el.matches(`.${dragClass} *`) && !el.classList.contains(dragClass)));
+
+    return cannotDrag;
+  }
+
+  private container: HTMLElement | undefined;
+  public setContainer(el: HTMLElement) {
+    this.container = el;
+  }
+
+  public getContainer() {
+    return this.container;
+  }
+
+  public getDropZones() {
+    if (!this.container) {
+      return [];
+    }
+
+    const layoutHasVisibleChildren =
+      [...this.container.children].filter((n) => Number.parseInt(n.getAttribute('data-order') ?? '0', 10) > -1).length >
+      0;
+    if (!layoutHasVisibleChildren) {
+      const child = this.container.appendChild(document.createElement('div'));
+      child.setAttribute('data-order', '99999');
+      const cells = calculateGridCells(this.container).filter((c) => c.order >= 0);
+      this.container.removeChild(child);
+      return cells;
+    }
+
+    return calculateGridCells(this.container).filter((c) => c.order >= 0);
   }
 }
 
 function SceneCSSGridLayoutRenderer({ model }: SceneCSSGridItemRenderProps<SceneCSSGridLayout>) {
   const { children, isHidden, isLazy } = model.useState();
-  const style = useLayoutStyle(model.state);
+  const styles = useStyles2(getStyles, model.state);
+  const containerRef = useRef<HTMLDivElement>(null);
+  const oldDropZone = useRef<DropZone>();
+  const { activeItem, activeLayout, dropZone } = model.dragManager.useState();
```

Comment:
```suggestion
  const { activeItem, activeLayout, dropZone } = model.dragManager.useState() ?? {};
```

I think we need something like this for cases where the drag manager isn't set. For example if I go to the demos listing page [here](http://localhost:3000/a/grafana-scenes-app/demos) it crashes because it expects the drag manager to exist

Reply:
You're right! Sorry, I was so focused on the demo app that I'd forgotten to check it didn't break other things. Will fix this asap